### PR TITLE
feat: add thousands separator

### PR DESCRIPTION
**Pull request by vlopezferrando**
Add thousands separator (`1_000_000`) to numbers (ints and floats, positive or negative).

Limitations:

* A max number of 4 underscores are added (the implementation has a copy-pasted regex, which may be improved with some kind of recursive logic).

### CODE REVIEW

**Review by morgante — APPROVED** [review]
I'm pretty sure we could do this generically without recursion, using either a more complex regex or a `split` of some sort.

File: .grit/patterns/thousands_separator.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+Add thousands separator (`1_000_000`) to numbers (ints and floats, positive or negative).
+
+```grit
+engine marzano(0.1)
+language python
+
+
+`$number` where {
+    $number <: or { integer(), float() },
```
Comment (morgante) [inline .grit/patterns/thousands_separator.md]: This is another example of the anti-pattern (maybe we should put this in the docs). If a variable is immediately restricted after assignment, start with the restriction.

```suggestion
or { integer(), float() } as number where {
```

**Review by morgante — CHANGES_REQUESTED** [review]

File: .grit/patterns/thousands_separator.md (hunk 1 above)
Comment (vlopezferrando) [reply .grit/patterns/thousands_separator.md]: Fixed.

**Comment by vlopezferrando** [comment]
Would it be better if the pattern just added one underscore?

```
1000 -> 1_000
1000000 -> 1000_000
1000_000 -> 1_000_000
```

Then, if applied multiple times, it would yield the right output no matter how long the number is.

**Review by morgante — APPROVED** [review]

**Pull request merged**